content/json-rpc.ts

```diff
@@ -101,40 +101,86 @@ class NSUser {
+   *
+   * @param terms  Single string as typed into the search box in Zotero (search for Title Creator Year)
+   *               Array of tuples similar as typed into the advanced search box in Zotero
+   *               (https://github.com/zotero/zotero/blob/9971f15e617f19f1bc72f8b24bb00b72d2a4736f/chrome/content/zotero/xpcom/data/searchConditions.js#L72-L610)
    */
-  public async search(terms: string | [string, string, string][], library?: string | number) {
+  public async search(terms: string
+  | ([string] | [string, string] | [string, string, string | number] | [string, string, string | number, boolean])[]) {
 
     const search = new Zotero.Search()
 
-    if (typeof terms === 'string') {
-      terms = terms.replace(/ (?:&|and) /g, ' ')
-      if (!/[\w\u007F-\uFFFF]/.test(terms)) return []
-      search.addCondition('quicksearch-titleCreatorYear', 'contains', terms)
-      search.addCondition('citationKey', 'contains', terms)
-    }
-    else {
-      if (!terms.length) return []
-      for (const [name, operator, value] of terms) {
-        search.addCondition(name, operator, value)
-        search.addCondition('citationKey', 'contains', value)
+    if (!terms.length) {/* */}
+    else if (typeof terms === 'string') {
+      // Custom action for only string.
+      // Similar behavior as quicksearch-titleCreateorYear, but search also in citationKey and ignore feeds and attachments
+
+      // Credit #2740
+      const fields = [
+        // search the quicksearch-titleCreatorYear fields
+        'title',
+        'publicationTitle',
+        'shortTitle',
+        'court',
+        'year',
+
+        // plus the citationKey
+        'citationKey',
+      ]
+
+      search.addCondition('blockStart')
+      for (const field of fields) {
+        search.addCondition(field, 'contains', terms, false)
       }
-    }
+      search.addCondition('blockEnd')
 
-    for (const feed of Zotero.Feeds.getAll()) {
-      search.addCondition('libraryID', 'isNot', feed.libraryID)
-    }
+      // Ignore Feeds
+      for (const feed of Zotero.Feeds.getAll()) {
+        search.addCondition('libraryID', 'isNot', feed.libraryID, true)
+      }
 
-    search.addCondition('itemType', 'isNot', 'attachment')
+      // Do not list attachments
+      search.addCondition('itemType', 'isNot', 'attachment', true)
 
-    if (typeof library !== 'undefined' && library !== '*') {
-      try {
-        search.addCondition('libraryID', 'is', Library.get(library).libraryID)
-      }
-      catch (err) {
-        throw new Error(`library ${JSON.stringify(library)} not found`)
+    }
+    else {
+      blk: for (const term of terms) {
+        // Custom Actions
+        if ((term.length === 1)) {
+          switch (term[0]) {
+            case 'ignore_feeds': {
+              for (const feed of Zotero.Feeds.getAll()) {
+                search.addCondition('libraryID', 'isNot', feed.libraryID)
```

Comment:
This should probably add `true` as the 4th parameter so these don't become an `OR` query if `joinMode` or `blockStart` are used.